src/common/utilities/Utils.ts

```diff
@@ -219,3 +219,84 @@ export function getVisibleCode(editor: vscode.TextEditor): { code: string; start
         endLine: firstVisibleRange.end.line
     };
 }
+
+
+async function getFileContent(activeFileUri: vscode.Uri, customExtension: string): Promise<{ customFileContent: string; customFileName: string; }> {
+    try {
+        const activeFileFolderPath = getFolderPathFromUri(activeFileUri);
+        const activeFileName = getFileNameFromUri(activeFileUri);
+
+        const activeFileNameParts = activeFileName.split('.');
+
+        let customFileName = activeFileNameParts[0];
+
+        for (let i = 1; i < activeFileNameParts.length - 2; i++) {
+            customFileName += `.${activeFileNameParts[i]}`;
+        }
+
+        customFileName += customExtension;
+
+        const customFilePath = path.join(activeFileFolderPath, customFileName);
+
+        // Read the content of the custom file
+        const diskRead = await import('fs');
+        const customFileContent = diskRead.readFileSync(customFilePath, 'utf8');
+
+        return { customFileContent, customFileName };
+    } catch (error) {
+        throw new Error(`Error reading the custom file content: ${error}`);
+    }
+}
+
+// Generic function to get file content based on type and component type
+async function getFileContentByType(activeFileUri: vscode.Uri, componentType: string, fileType: string): Promise<{ customFileContent: string; customFileName: string; }> {
+    try {
+        const extension = componentTypeSchema[componentType]?.[fileType];
+        if (!extension) {
+            throw new Error(`File type ${fileType} not found for component type ${componentType}`);
+        }
+        return await getFileContent(activeFileUri, extension);
+    } catch (error) {
+        const message = (error as Error)?.message;
```

Comment:
what experiene will user have in case of error like this?

Reply:
User's will not see this error, the related files will not be passed to LLM and there will be variation in response.